lib/addons/karpenter/index.ts

```diff
@@ -0,0 +1,137 @@
+import { Construct } from '@aws-cdk/core';
+import { Role, ManagedPolicy, ServicePrincipal, CfnInstanceProfile, PolicyDocument } from '@aws-cdk/aws-iam';
+import { ClusterInfo } from '../../spi';
+import { HelmAddOn, HelmAddOnProps, HelmAddOnUserProps } from '../helm-addon';
+import { createNamespace, setPath, createServiceAccount, conflictsWith, tagSubnets } from '../../utils'
+import { KarpenterControllerPolicy } from './iam'
+
+/**
+ * Configuration options for the add-on
+ */
+interface KarpenterAddOnProps extends HelmAddOnUserProps {
+    /**
+     * Specs for a Provisioner (Optional)
+     */
+    ProvisionerSpecs?: { 
+        'node.kubernetes.io/instance-type': string[],
+        'topology.kubernetes.io/zone': string[],
+        'kubernetes.io/arch': string[],
+        'karpenter.sh/capacity-type': string[],
+    }
+}
+
+const KARPENTER = 'karpenter'
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps: HelmAddOnProps = {
+    name: KARPENTER,
+    namespace: KARPENTER,
+    version: '0.5.3',
+    chart: KARPENTER,
+    release: "ssp-addon-karpenter",
+    repository: 'https://charts.karpenter.sh',
+}
+
+/**
+ * Implementation of the Karpenter add-on
+ */
+export class KarpenterAddOn extends HelmAddOn {
+
+    readonly options: KarpenterAddOnProps;
+
+    constructor(props?: KarpenterAddOnProps) {
+        super({...defaultProps, ...props});
+        this.options = this.props;
+    }
+
+    @conflictsWith('ClusterAutoScalerAddOn')
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const endpoint = clusterInfo.cluster.clusterEndpoint
+        const name = clusterInfo.cluster.clusterName
+        const cluster = clusterInfo.cluster
+        const values = { ...this.props.values ?? {} }
+
+        // Tag VPC Subnets
+        tagSubnets(cluster.stack, cluster.vpc.privateSubnets, `kubernetes.io/cluster/${clusterInfo.cluster.clusterName}`,"1")
+
+        // Set up Node Role
+        const karpenterNodeRole = new Role(cluster, 'karpenter-node-role', {
+            assumedBy: new ServicePrincipal(`ec2.${cluster.stack.urlSuffix}`),
+            managedPolicies: [
+                ManagedPolicy.fromAwsManagedPolicyName("AmazonEKSWorkerNodePolicy"),
```

Comment:
Why do we need these managed policies along with Karpenter controller role? Can you link the documentation on the required IAM configuration and reference it in the docs section?

Reply:
Documentation from Karpenter site is linked. It explains what is needed and why it's needed, as well as the yaml file that references the policies required.